src/dashboards/videos/details.jsonnet

```
@@ -1,17 +1,118 @@
 local grafana = import 'grafonnet/grafana.libsonnet';
 local dashboard = grafana.dashboard;
+local elasticsearch = grafana.elasticsearch;
+local graphPanel = grafana.graphPanel;
 local template = grafana.template;
 
+local lrs = 'lrs';
+local course_field = 'object.definition.extensions.http://adlnet.gov/expapi/activities/course.keyword';
+local school_field = 'object.definition.extensions.https://w3id.org/xapi/acrossx/extensions/school.keyword';
+local session_field = 'object.definition.extensions.http://adlnet.gov/expapi/activities/module.keyword';
+local video_id_field = 'object.id.keyword';
+
 dashboard.new(
   'Details',
   tags=['xAPI', 'video', 'teacher'],
   editable=false
 )
 .addTemplate(
-  template.datasource(
-    'PROMETHEUS_DS',
-    'prometheus',
-    'Prometheus',
-    hide='label',
+  template.new(
+    name='SCHOOL',
+    current='all',
+    label='School',
+    datasource=lrs,
+    query='{"find": "terms", "field": "%(school)s"}' % {
+      school: school_field,
+    },
+    refresh='time'
+  )
+)
+.addTemplate(
+  template.new(
+    name='COURSE',
+    current='all',
+    label='Course',
+    datasource=lrs,
+    // FIXME: We perform a global search here because we are unable to escape
+    // the ':' character from the school field name.
+    //
+    // Possible workaround: add alias fields to the ES index, e.g.
+    // course -> object.definition.extensions.http://adlnet.gov/expapi/activities/course.keyword
+    query='{"find": "terms", "field": "%(course)s", "query": "$SCHOOL"}' % {
+      course: course_field,
```

Escaping it twice seems to work. Could you give it a try?

```suggestion
    query='{"find": "terms", "field": "%(course)s", "query": "%(school)s:$SCHOOL"}' % {
      course: course_field,
      school: double_escape_string(school_field),
```
Where:
`local double_escape_string(x) = std.strReplace(std.strReplace(x, ':', '\\\\:'), '/', '\\\\/');`

---

And it works, thanks! :pray: